server/handler.go

```diff
@@ -153,6 +154,7 @@ func (h *Handler) ComQuery(
 	}
 
 	start := time.Now()
+	parsedQuery, parseErr := sqlparser.Parse(query)
```

Comment:
This seems weird.  If there is a parsing error we continue doing everything like normal until determining if it should be committed?  If there is a parse error shouldn't everything between here and where the error is handled be skipped?

Reply:
No, because the engine has their own additional parsing logic outside of vitess's parser. I'm adding a comment.